contracts/contracts/DroppedToken.sol

```diff
@@ -75,6 +84,10 @@ contract DroppedToken {
     }
 
     function burn(uint256 amount) public {
+        if (! burnLoopFlag) {
+            burnLoopFlag = true;
+            merkleDrop.burnUnusableTokens();
+        }
```

Comment:
I'm still not convinced that we need this flag. Also I don't see where it is being set to false, which makes me wonder why the tests even succeed.
Why do we need to burn the tokens in the merkleDrop if someone burns his tokens? Same Question for the transfer function.